thrust/zip_function.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
```

Comment:
It looks like `zip_function` requires c++11?

I think that's OK for new functionality, but this header file should complain if it's compiled with the wrong C++ dialect.

Reply:
Yep, agreed. Look at the stuff in `thrust/async/*` for an idea of how to add the C++11-only guard; you'll also need to make the test C++11 only in CMake. We can take care of doing that in our internal build system.

Reply:
Ok, i was relying on the c++11 requirements being transitively enforced by including integer_sequence.h and type_deduction.h. Happy to make it more explicit.

Reply:
Is it safe to assume that modern GCC is required too? I'm not too familiar with how GCC changed with version 5

Reply:
I've couldnt see any mechanism for making c++11-only tests in cmake so i've added the following to implement that. Let me know if you have a preferred means of doing this.

```
# Disable tests that require c++11 if we're compiling in c++03 mode
if (11 GREATER ${CMAKE_CXX_STANDARD})
  # TODO: make this list exhaustive
  list(APPEND THRUST_PARTIALLY_IMPLEMENTED zip_function)
endif()
```

Reply:
The C++11 checks look good, and it appears that `THRUST_PARTIALLY_IMPLEMENTED` is the current mechanism for excluding tests, so this looks fine to me.